service/history/execution/context_util.go

```diff
@@ -30,6 +31,52 @@ import (
 	"github.com/uber/cadence/common/types"
 )
 
+func (c *contextImpl) emitLargeWorkflowShardIDStats() {
+	if c.shard.GetConfig().EnableShardIDMetrics() {
+		shardIDStr := strconv.Itoa(c.shard.GetShardID())
+
+		var blobSizeWarn int64
+		if c.shard.GetConfig().LargeShardHistoryBlobMetricThreshold() < c.shard.GetConfig().BlobSizeLimitWarn(c.GetDomainName()) {
+			blobSizeWarn = int64(c.shard.GetConfig().LargeShardHistoryBlobMetricThreshold())
+		} else {
+			blobSizeWarn = int64(c.shard.GetConfig().BlobSizeLimitWarn(c.GetDomainName()))
+		}
```

Comment:
just do it once and get the min of both values.  dynamic config can have arbitrary behavior under it, it's not (necessarily) free.